extensions/amp-sidebar/1.0/storybook/Basic.js

```diff
@@ -80,6 +80,33 @@ export const _default = () => {
   );
 };
 
+export const toolbar = () => {
+  const sideConfigurations = ['left', 'right', undefined];
+  const side = select('side', sideConfigurations, sideConfigurations[0]);
+  const toolbarMedia = text('toolbar media', '(max-width: 500px)');
+  const foregroundColor = color('color');
+  const backgroundColor = color('background');
+  const backdropColor = color('backdrop color');
+
+  return (
+    <main>
+      <SidebarWithActions
+        side={side}
+        style={{color: foregroundColor, backgroundColor}}
+        backdropStyle={{backgroundColor: backdropColor}}
+      >
+        <SidebarToolbar toolbar={toolbarMedia} toolbar-target="toolbar-target">
+          <ul>
+            <li>Toolbar Item 1</li>
+            <li>Toolbar Item 2</li>
+          </ul>
+        </SidebarToolbar>
+      </SidebarWithActions>
+      <div id="toolbar-target"></div>
+    </main>
+  );
```

Comment:
This is the `toolbarTarget` as `ref` idea:
```suggestion
  const targetRef = useRef(null);

  return (
    <main>
      <SidebarWithActions
        side={side}
        style={{color: foregroundColor, backgroundColor}}
        backdropStyle={{backgroundColor: backdropColor}}
      >
        <SidebarToolbar toolbar={toolbarMedia} toolbarTarget={targetRef}>
          <ul>
            <li>Toolbar Item 1</li>
            <li>Toolbar Item 2</li>
          </ul>
        </SidebarToolbar>
      </SidebarWithActions>
      <div ref={targetRef}></div>
    </main>
  );
```

Then your `createPortal` line would look like:
`{matches && toolbarTarget.current && createPortal(children, toolbarTarget.current)}`

I'm actually not sure if it's advisable to pass a ref like this, and usually indicates that some outer context may be helpful to manage things more explicitly... But the current approach of querying the document doesn't allow for updating the portalled parent i.e. if the same id gets reassigned to a different element between renders.

Reply:
I see, I'm open to it either way.  Let's see if anyone else feels strongly about it.  If not, we can table it for now and explore further as a follow up.